data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/HashRing.java

```diff
@@ -50,12 +49,12 @@ public HashRing(final PeerListProvider peerListProvider, final int numVirtualNod
         peerListProvider.addListener(this);
     }
 
-    public Optional<String> getServerIp(final Set<String> identificationKeys) {
+    public Optional<String> getServerIp(final List<String> identificationKeyValues) {
```

Comment:
What's the reason for this update to `List`?

Reply:
The values can be sometimes be same, for example `[value1, value1]`. So we did't want to ignore the second value while calculating hash.